File: src/lib/api/stock.ts
import { invoke } from "./client";

export type EstadoStock = "sin_stock" | "bajo" | "ok";

export type ProductoStock = {
  id: number;
  codigoInterno: string;
  nombre: string;
  marcaNombre: string | null;
  categoriaNombre: string | null;
  stockActual: number;
  stockMinimo: number;
  estadoStock: EstadoStock;
};

export function listarStock() {
  return invoke<ProductoStock[]>("stock_listar");
}

export function listarReposicion() {
  return invoke<ProductoStock[]>("stock_listar_reposicion");
}

/** Único movimiento posible en la Fase 4: ajuste manual con motivo obligatorio. */
export function ajustarStock(productoId: number, nuevaCantidad: number, motivo: string) {
  return invoke<void>("stock_ajustar", {
    productoId,
    datos: { nuevaCantidad, motivo },
  });
}

export function actualizarStockMinimo(productoId: number, stockMinimo: number) {
  return invoke<void>("stock_actualizar_minimo", { productoId, stockMinimo });
}

File: src-tauri/src/models/stock.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, sqlx::FromRow)]
#[serde(rename_all = "camelCase")]
pub struct ProductoStock {
    pub id: i64,
    pub codigo_interno: String,
    pub nombre: String,
    pub marca_nombre: Option<String>,
    pub categoria_nombre: Option<String>,
    pub stock_actual: i64,
    pub stock_minimo: i64,
    /// "sin_stock" | "bajo" | "ok" -- calculado en la consulta, no guardado.
    pub estado_stock: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AjusteStock {
    pub nueva_cantidad: i64,
    pub motivo: String,
}
